### DISCUSSION TITLE
Different numbers of clones in previous mixcr version (3) and new version (4.1.2)

### DISCUSSION THREAD

**Discussion by AnkaShch** [milaboratory/mixcr#994] [Q&A] [answered]
Hello!

We are trying to launch a new version of mixcr on the same data that we had calculated on version 3. We are using [this tutorial](https://docs.milaboratories.com/mixcr/guides/rnaseq/) and have a few question:
1. Why we have different numbers of clones in BCR (for mixcr v4.1.2 we summarize number of rows in _clones_IGH.tsv_, _clones_IGK.tsv_ and _clones_IGL.tsv_ and have 455 rows, for mixcr v3 we have 873)?
2. Are we understood right that tutorial not fully update? We have different resulting files.
3. Why in tutorial said that it is not work for BCR, but we have results for them?

Thank you for your answering!

**Comment by mizraelson** [comment]
Hi,
Can you please share the exact commands you use for MiXCR v3 and 4.

The tutorial might be slightly outdated, I apologize for that, I'll fix it.

Also, to answer the question about tcr and bcr data analysis. In all cases MiXCR tries to align reads against both TCR and BCR references. However, we have different aligners for BCR data, which suits it better. TCR aligner is not tuned to work with BCR data, but it's faster, BCR aligner works with both TCR and BCR, but it is more time consuming. Regardless of the aligner you will get both TCR and BCR clones, unless specified otherwise on the export step.

**Reply by AnkaShch** [reply]
Thank you!

For v4 we used same command as in tutorial:
```
mixcr analyze rnaseq-cdr3 \
--species hsa \
rna_1.fastq.gz  rna_2.fastq.gz \
results/
```
and for v3 we have not exact command, sorry, we calculated it a long time ago. But parameters were be something like:
```
mixcr align -p rna-seq --aspecies hsa -OallowPartialAlignments=true -OsaveOriginalReads=true
mixcr assamble -OaddReadsCountOnClustering=true -ObadQualityThreshold=10 --write-alignments -OseparateByC=true
mixcr export -p fullImputed
```

**Comment by mizraelson** [comment, marked as answer]
There are few reasons why you might get different results:

1) I see you use `--write-alignments` on assemble and `-p fullImputed` on export. Does that mean that you are trying to reconstruct the full-length receptor sequence (not just CDR3 region)? If that's the case I think you also used `mixcr assembleContigs` step for MiXCR v3 (the one that uses the alignments you preserve with `--write-alignments` on assemble step to reconstruct the full length sequence).

To reproduce similar behavior you want to use the following preset: `rnaseq-full-length`.

2) You used `-ObadQualityThreshold=10` for MiXCR v3 and in MiXCR v4 this parameter is set to 20 for the RNA-Seq preset. 
You can try adding the following parameter to your command to overwrite that default value:

`-Massemble.cloneAssemblerParameters.badQualityThreshold=10`

3) If you want to separate clones by C gene  you can add:

`--split-clones-by C`

4) To impute uncovered parts of the regions from the germ-line, use the following parameter:

`--impute-germline-on-export`

So you will get something like:

```
mixcr analyze rnaseq-full-length \
--species hsa \
-Massemble.cloneAssemblerParameters.badQualityThreshold=10 \
--split-clones-by C \
--impute-germline-on-export \ 
rna_1.fastq.gz  rna_2.fastq.gz \
results/
```

**Reply by AnkaShch** [reply]
Thank you, now it is clear!
